### PR TITLE
add stm32l4r9 bsp 

### PR DESCRIPTION
**Pull request by jhbdream**
## 拉取/合并请求描述：(PR description)

[
    为L4系列 [drv_gpio.c] 添加 VDDIO2 使能
    为STM32L4R9-ST-EVAL 添加LCD驱动支持
    为STM32L4R9-ST-EVAL 添加TOUCH驱动支持
    修改其他错误
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other style
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [ ] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

File: bsp/stm32/libraries/HAL_Drivers/drv_gpio.c [hunk 1]
```diff
@@ -755,6 +755,9 @@ int rt_hw_pin_init(void)
 #endif
 
 #if defined(__HAL_RCC_GPIOG_CLK_ENABLE)
+	#ifdef SOC_SERIES_STM32L4
```
Comment (BernardXiong) [inline bsp/stm32/libraries/HAL_Drivers/drv_gpio.c]: Please take care of indentation.

File: bsp/stm32/stm32l4r9-st-eval/board/SConscript [hunk 2]
```diff
@@ -12,8 +12,20 @@ board.c
 CubeMX_Config/Src/stm32l4xx_hal_msp.c
 ''')
 
+
+if GetDepend(['BSP_USING_DSI']):
+	src += Glob('ports/drv_lcd_dsi.c')
```
Comment (BernardXiong) [inline bsp/stm32/stm32l4r9-st-eval/board/SConscript]: indentation issue...

File: bsp/stm32/stm32l4r9-st-eval/board/ports/drv_touch.c [hunk 3]
```diff
@@ -0,0 +1,208 @@
+/*
+ * File      : drv_touch.c
+ * This file is part of RT-Thread RTOS
```
Comment (BernardXiong) [inline bsp/stm32/stm32l4r9-st-eval/board/ports/drv_touch.c]: Please take care of license issue. These files should be license under Apache License v2.0

File: bsp/stm32/libraries/HAL_Drivers/SConscript [hunk 4]
```diff
@@ -27,7 +27,8 @@ if GetDepend(['RT_USING_QSPI']):
     src += ['drv_qspi.c']
 
 if GetDepend(['RT_USING_I2C', 'RT_USING_I2C_BITOPS']):
-    src += ['drv_soft_i2c.c']
+	if  GetDepend('BSP_USING_I2C1') or GetDepend('BSP_USING_I2C2') or GetDepend('BSP_USING_I2C3') or GetDepend('BSP_USING_I2C4'):
```
Comment (armink) [inline bsp/stm32/libraries/HAL_Drivers/SConscript]: tab to 4spaces

**Comment by jhbdream** [comment]
使用的 notepad++ 编辑器，他的缩进会有些问题

File: bsp/stm32/stm32l4r9-st-eval/board/SConscript [hunk 5]
```diff
@@ -12,8 +12,19 @@ board.c
...
 ''')
 
+if GetDepend(['BSP_USING_DSI']):
+    src += Glob('ports/drv_lcd_dsi.c')
+
+if GetDepend(['BSP_USING_SRAM']):
+    src += Glob('ports/drv_sram.c')
+
+if GetDepend(['BSP_USING_TOUCH']):
+	src += Glob('ports/drv_touch.c')
```
Comment (BernardXiong) [inline bsp/stm32/stm32l4r9-st-eval/board/SConscript]: indentation issue.

File: bsp/stm32/stm32l4r9-st-eval/board/ports/drv_touch_ft.c [hunk 6]
```diff
@@ -0,0 +1,232 @@
+/*
+ * File      : drv_touch_ft.c
+ *             ft touch driver
+ * COPYRIGHT (C) 2006 - 2017, RT-Thread Development Team
```
Comment (BernardXiong) [inline bsp/stm32/stm32l4r9-st-eval/board/ports/drv_touch_ft.c]: license issue.

File: bsp/stm32/stm32l4r9-st-eval/board/ports/include/drv_touch.h [hunk 7]
```diff
@@ -0,0 +1,68 @@
+/*
+ * File      : drv_touch.h
+ * This file is part of RT-Thread RTOS
+ * COPYRIGHT (C) 2017, RT-Thread Development Team
```
Comment (BernardXiong) [inline bsp/stm32/stm32l4r9-st-eval/board/ports/include/drv_touch.h]: license issue.

**Comment by BernardXiong** [comment]
1. 几次修改合并后进行一次性提交；
2. git commit -m 'message' 中的message需要有一定的规范性，这部分在issue中已经有人专门提及了；

重新提交，并把规范处理好。

**Pull request closed without merging**